Static binary assets are not served correctly when they have capital letters in extension

When serving static binary assets configured with:
```javascript
files: [
	...
	{pattern: './fixtures/images/*.jpg', included: false, served: true},
	{pattern: './fixtures/images/*.JPG', included: false, served: true},
	...
]
```
Only images with *.jpg extensions are served properly. Capital letters extension shows a 'broken image' icon. This seems to be due to not recognising said image as a binary file. Adding "JPG" to `lib/binary-extensions.json` or modifying `lib/preprocessor.js` condition like so:
```javascript
var thisFileIsBinary = isBinary[path.extname(file.originalPath).toLowerCase()]
```
Solves the problem.

Not sure how to write a failing test for this to submit a fix.

Current `master` branch has failing tests for me before I even started editing both under node v0.12.6 and iojs v2.3.3:
```
$ grunt
Running "browserify:client" (browserify) task
>> Bundle static/karma.js created.

Running "test:unit" (test) task

Running "mochaTest:unit" (mochaTest) task


  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․

  377 passing (5s)
  3 pending


Running "test:client" (test) task
Running /Users/kruk/dev/karma/bin/karma start test/client/karma.conf.js --single-run --no-auto-watch --reporters=dots
failed to add custom browserify preprocessor
{ [Error: Cannot find module 'karma/lib/preprocessor'] code: 'MODULE_NOT_FOUND' }
22 07 2015 00:05:14.797:INFO [framework.browserify]: 472361 bytes written (0.87 seconds)
22 07 2015 00:05:14.808:INFO [karma]: Karma v0.13.2 server started at http://localhost:9876/
22 07 2015 00:05:14.814:INFO [launcher]: Starting browser Chrome
22 07 2015 00:05:16.157:INFO [Chrome 43.0.2357 (Mac OS X 10.10.4)]: Connected on socket rTnfwhUtnLPRh4G1AAAA with id 19490884
Chrome 43.0.2357 (Mac OS X 10.10.4) ERROR
  Uncaught ReferenceError: require is not defined
  at /Users/kruk/dev/karma/test/client/karma.spec.js:1
Chrome 43.0.2357 (Mac OS X 10.10.4): Executed 0 of 0 ERROR (0.162 secs / 0 secs)
[Error: failed to add custom browserify preprocessor
{ [Error: Cannot find module 'karma/lib/preprocessor'] code: 'MODULE_NOT_FOUND' }]
Fatal error: Client unit tests failed.
```

--- a/lib/preprocessor.js
+++ b/lib/preprocessor.js
@@ -47,7 +47,7 @@ var createPreprocessor = function (config, basePath, injector) {
 
   return function preprocess (file, done) {
     patterns = Object.keys(config)
-    var thisFileIsBinary = isBinary[path.extname(file.originalPath)]
+    var thisFileIsBinary = isBinary[path.extname(file.originalPath).toLowerCase()]
     var preprocessors = []
 
     var nextPreprocessor = function (error, content) {

--- a/test/unit/preprocessor.spec.coffee
+++ b/test/unit/preprocessor.spec.coffee
@@ -14,6 +14,7 @@ describe 'preprocessor', ->
         'b.js': mocks.fs.file 0, 'content'
         'a.txt': mocks.fs.file 0, 'some-text'
         'photo.png': mocks.fs.file 0, 'binary'
+        'CAM_PHOTO.JPG': mocks.fs.file 0, 'binary'
 
     mocks_ =
       'graceful-fs': mockFs
@@ -190,6 +191,24 @@ describe 'preprocessor', ->
     pp = m.createPreprocessor {'**/*': ['fake']}, null, injector
 
     file = {originalPath: '/some/photo.png', path: 'path'}
+
+    pp file, (err) ->
+      expect(fakePreprocessor).not.to.have.been.called
+      expect(file.content).to.be.an.instanceof Buffer
+      done()
+
+
+  it 'should not preprocess binary files with capital letters in extension', (done) ->
+    fakePreprocessor = sinon.spy (content, file, done) ->
+      done null, content
+
+    injector = new di.Injector [{
+      'preprocessor:fake': ['factory', -> fakePreprocessor]
+    }]
+
+    pp = m.createPreprocessor {'**/*': ['fake']}, null, injector
+
+    file = {originalPath: '/some/CAM_PHOTO.JPG', path: 'path'}
 
     pp file, (err) ->
       expect(fakePreprocessor).not.to.have.been.called

